classes/class-settings.php

```diff
@@ -471,6 +471,50 @@ public function get_defaults() {
 		return (array) $defaults;
 	}
 
+	/**
+	 * Retrieves the deletion warning message based on the site type
+	 * and whether or not there is currently a process running to delete the tables.
+	 *
+	 * @return string The deletion warning message.
+	 */
+	public function get_deletion_warning(): string {
+		$site_type = $this->plugin->get_site_type();
+
+		switch ( $site_type ) {
+			case Admin::WP_STREAM_MULTI_NETWORK:
+				$warning = __( 'Warning: This will delete all activity records from the database for all sites.', 'stream' );
+				break;
+			case Admin::WP_STREAM_MULTI_NOT_NETWORK:
+				$warning = $this->plugin->is_large_records_table( Admin::get_blog_record_table_size() ) ?
+						$this->get_async_deletion_warning() :
+						__( 'Warning: This will delete all activity records from the database for this site.', 'stream' );
+				break;
+			default:
+				$warning = __( 'Warning: This will delete all activity records from the database.', 'stream' );
+				break;
+		}
+
+		return $warning;
+	}
+
+	/**
+	 * Retrieves the warning message for asynchronous deletion.
+	 *
+	 * This method checks if there is an action scheduler event running already deleting things
+	 * and returns an appropriate message.
+	 *
+	 * @return string The warning message.
+	 */
+	private function get_async_deletion_warning() {
+		// Check if there is an action scheduler event running already deleting things.
+		$has_scheduled_action = Admin::is_running_async_deletion();
+
+		return $has_scheduled_action ?
+			__( 'Currently deleting records. Please be patient, this can take a while.', 'stream' )
+			:
+			__( 'Warning: This will delete all activity records from the database for this site. If any are added while this is running, they will not be deleted.', 'stream' );
+	}
+
```

Comment:
<!-- trivial preference-based requests -->
**🥜 nitpick:** A very minor thing yet I'd like to share with you. I find this code snippet relatively hard to read because of this ternaries and indirection between them (one ternary calls a method which has yet another ternary inside). Do you think it'd be clearer if the `get_async_deletion_warning()` logic was moved over to the switch statement and the regular `if .. else` statements used instead of ternaries?

Reply:
@delawski I've updated it here to use `if ... else` statements and checked for the AS action first as I think that's the most robust for future development (in case anything else uses the async deletion, we don't have to worry about the conditional here) plus it's a cheap check. Is that ok?

Also I added more empty lines than usual as I have a bit of trouble seeing if / else statements when there are no empty lines for some reason.

Commit https://github.com/xwp/stream/pull/1543/commits/12ca1909854decc9e0326998e4cf884f890ea487

Reply:
This looks great, thank you!

> Also I added more empty lines than usual as I have a bit of trouble seeing if / else statements when there are no empty lines for some reason.

All good! 😄 I wonder if it would make it easier to read for you if there were only `if` statements (without any `elseif`) and they simply returned the warning message directly (there's no other logic besides that `if .. else` in the function). Anyway, let's leave it as is, it's super clean and readable to me now 👍